src/api/utils/coingecko.ts

```diff
@@ -0,0 +1,247 @@
+/*
+  Copyright 2021 Set Labs Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+'use strict';
+
+const pageResults = require('graph-results-pager');
+
+import axios from 'axios';
+import Assertions from '../../assertions';
+
+import {
+  CoinGeckoCoinPrices,
+  CoinGeckoTokenData,
+  SushiswapTokenData,
+  CoinGeckoTokenMap,
+  CoinPricesParams,
+  PolygonMappedTokenData
+} from '../../types';
+
+/**
+ * These currency codes can be used for the vs_currencies parameter of the service's
+ * fetchCoinPrices method
+ *
+ * @type {number}
+ */
+export const USD_CURRENCY_CODE = 'usd';
+export const ETH_CURRENCY_CODE = 'eth';
+
+/**
+ * @title CoinGeckoDataService
+ * @author Set Protocol
+ *
+ * A utility library for fetching token metadata and coin prices from Coingecko for Ethereum
+ * and Polygon chains
+ */
+export class CoinGeckoDataService {
+  chainId: number;
+  private tokenList: CoinGeckoTokenData[] | undefined;
+  private tokenMap: CoinGeckoTokenMap | undefined;
+  private assert: Assertions;
+
+  constructor(chainId: number) {
+    this.assert = new Assertions();
+    this.assert.common.isSupportedChainId(chainId);
+    this.chainId = chainId;
+  }
+
+  /**
+   * Gets address-to-price map of token prices for a set of token addresses and currencies
+   *
+   * @param  params             CoinPricesParams: token addresses and currency codes
+   * @return                    CoinGeckoCoinPrices: Address to price map
+   */
+  async fetchCoinPrices(params: CoinPricesParams): Promise<CoinGeckoCoinPrices> {
+    const platform = this.getPlatform();
```

Comment:
What happens on mumbai or kovan? Should return early with array of 0s?

Reply:
Atm unsupported platforms throw with an error.  I think that's might be the right behavior ... the price apis aren't meaningful on those networks.